flink-libraries/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -173,55 +180,92 @@ public void open() {
 			if (line == null || line.equals("")) {
 				continue;
 			}
+			parseAndCall(line);
+		}
+	}
 
-			final SqlCommandCall cmdCall = SqlCommandParser.parse(line);
+	/**
+	 * Submits a SQL update statement and prints status information and/or errors on the terminal.
+	 *
+	 * @param statement SQL update statement
+	 * @return flag to indicate if the submission was successful or not
+	 */
+	public boolean submitUpdate(String statement) {
```

Comment:
do we have tests for that? Some `ITCase`?

Reply:
The entire CliClient is only tested manually so far.

Reply:
Hmmm, that's bad :( Is it hard to start local flink mini cluster and execute `cli` process from `JUnit` test? (Maybe as a follow up after feature freeze)

Reply:
I added some unit tests.